### ISSUE TITLE
Generates NumberFormatException for non-number columns

### ISSUE THREAD

**Issue by rnix** [orientechnologies/orientdb-etl#46]
Example failures:

This is an address, not a number:

    [96681:csv] ERROR Error on setting document field ADDRESS=null (cause=java.lang.NumberFormatException: For input string: "11 Greenway Plaza Suite 2625")

This is a date, not a number:

    [96681:csv] ERROR Error on setting document field ESTYMD=null (cause=java.lang.NumberFormatException: For input string: "2008-12-31")

**Comment by lvca** [comment]
I think it's trying guessing the type. As workaround, could you create a schema assigning types on properties?

**Comment by lvca** [comment]
Do you have also the source record and JSON cfg to reproduce it?

**Comment by rnix** [comment]
The workaround isn't useful to me. I'm looking into this process to avoid that step of our current ETL process.

The offending row of `test.csv`:

	ADDRESS,BKCLASS,CBSA,CBSA_DIV,CBSA_DIV_FLG,CBSA_DIV_NO,CBSA_METRO,CBSA_METRO_FLG,CBSA_METRO_NAME,CBSA_MICRO_FLG,CBSA_NO,CERT,CITY,COUNTY,CSA,CSA_FLG,CSA_NO,ESTYMD,FI_UNINUM,MAINOFF,NAME,OFFNAME,OFFNUM,RUNDATE,SERVTYPE,STALP,STCNTY,STNAME,UNINUM,ZIP
	11 Greenway Plaza Suite 2625,N,"Houston-The Woodlands-Sugar Land, TX", ,0,0,26420,1,"Houston-The Woodlands-Sugar Land, TX",0,26420,91325,Houston,Harris,"Houston-The Woodlands, TX",1,288,2008-12-31,18316,0,"Atlantic Trust Company, National Association",Atlantic Trust Company Houston Branc,7,2015-03-26,30,TX,48201,Texas,494874,77046
	"401 Congress Ave, Suite 2450",N,"Austin-Round Rock, TX", ,0,0,12420,1,"Austin-Round Rock, TX",0,12420,91325,Austin,Travis, ,0,,2008-12-31,18316,0,"Atlantic Trust Company, National Association",Atlantic Trust Company Austin Branch,4,2015-03-26,30,TX,48453,Texas,494871,78701

And the JSON `test.json`:

    {
      "config": {
        "log": "debug"
      },
      "begin": [
      ],
      "source" : {
        "file": { "path": "/tmp/test.csv", "lock" : true }
      },
      "extractor" : {
        "row": {}
      },
      "transformers" : [
       { "csv": { "separator": ",", "nullValue": "NULL", "skipFrom": 1, "skipTo": 1 } },
       { "vertex": { "class": "bank"} }
      ],
      "loader" : {
        "orientdb": {
          "dbURL": "remote:localhost/test",
          "dbUser": "admin",
          "dbPassword": "admin",
          "dbAutoCreate": true,
          "tx": false,
          "batchCommit": 1000,
          "dbType": "graph"
        }
      }
    }

**Comment by kaidzen** [comment]
Hello,
I'd tried to reproduce this error and got this output:
```
[2:csv] ERROR Error on setting document field ESTYMD=null (cause=java.lang.NumberFormatException: For input string: "2008-12-31")
[2:csv] ERROR Error on setting document field RUNDATE=null (cause=java.lang.NumberFormatException: For input string: "2015-03-26")
```

There is a problem with type of data determinations in OCSVTransformer.java:141
```java
        else if (fieldStringValue != null && !fieldStringValue.isEmpty()) {
          // DETERMINE THE TYPE
          final char firstChar = fieldStringValue.charAt(0);
          if (firstChar == stringCharacter)
            // STRING
            fieldValue = OStringSerializerHelper.getStringContent(fieldStringValue);
          else if (Character.isDigit(firstChar))
            // NUMBER
            if (fieldStringValue.contains(".") || fieldStringValue.contains(","))
              try {
                fieldValue = Float.parseFloat(fieldStringValue);
              } catch (Exception e) {
                fieldValue = Double.parseDouble(fieldStringValue);
              }
            else
              try {
                fieldValue = Integer.parseInt(fieldStringValue);
              } catch (Exception e) {
                fieldValue = Long.parseLong(fieldStringValue);
              }
```
As you see, there is no implementation for DATE TYPE resolving.

I would like to fix this error and will send you PR

**Comment by rnix** [comment]
@kaidzen I wonder if you try the import with the lines flipped, does it produce a different error?

    ADDRESS,BKCLASS,CBSA,CBSA_DIV,CBSA_DIV_FLG,CBSA_DIV_NO,CBSA_METRO,CBSA_METRO_FLG,CBSA_METRO_NAME,CBSA_MICRO_FLG,CBSA_NO,CERT,CITY,COUNTY,CSA,CSA_FLG,CSA_NO,ESTYMD,FI_UNINUM,MAINOFF,NAME,OFFNAME,OFFNUM,RUNDATE,SERVTYPE,STALP,STCNTY,STNAME,UNINUM,ZIP
    "401 Congress Ave, Suite 2450",N,"Austin-Round Rock, TX", ,0,0,12420,1,"Austin-Round Rock, TX",0,12420,91325,Austin,Travis, ,0,,2008-12-31,18316,0,"Atlantic Trust Company, National Association",Atlantic Trust Company Austin Branch,4,2015-03-26,30,TX,48453,Texas,494871,78701
    11 Greenway Plaza Suite 2625,N,"Houston-The Woodlands-Sugar Land, TX", ,0,0,26420,1,"Houston-The Woodlands-Sugar Land, TX",0,26420,91325,Houston,Harris,"Houston-The Woodlands, TX",1,288,2008-12-31,18316,0,"Atlantic Trust Company, National Association",Atlantic Trust Company Houston Branc,7,2015-03-26,30,TX,48201,Texas,494874,77046

**Comment by kaidzen** [comment]
Error fixed in PL #48 
@rnix I'd tested your file, and it imported successfully.

**Comment by lvca** [comment]
Thanks, merged ;-)
